source/class/qx/tool/cli/config/CompilerConfig.js

```diff
@@ -0,0 +1,111 @@
+/* ************************************************************************
+ *
+ *    qooxdoo-compiler - node.js based replacement for the Qooxdoo python
+ *    toolchain
+ *
+ *    https://github.com/qooxdoo/qooxdoo-compiler
+ *
+ *    Copyright:
+ *      2011-2019 Zenesis Limited, http://www.zenesis.com
+ *
+ *    License:
+ *      MIT: https://opensource.org/licenses/MIT
+ *
+ *      This software is provided under the same licensing terms as Qooxdoo,
+ *      please see the LICENSE file in the Qooxdoo project's top-level directory
+ *      for details.
+ *
+ *    Authors:
+ *      * John Spackman (john.spackman@zenesis.com, @johnspackman)
+ *
+ * *********************************************************************** */
+
+const path = require("path");
+const fs = qx.tool.utils.Promisify.fs;
+
+/**
+ * Loads and stores the configuration for the entire compiler
+ * 
+ */
+qx.Class.define("qx.tool.cli.config.CompilerConfig", {
+  extend: qx.tool.cli.config.AbstractConfig,
+  
+  construct: function(command) {
+    this.base(arguments);
+    this.__command = command;
+    this.__libraryConfigs = {};
+  },
+  
+  properties: {
+    configFilename: {
+      init: "compile.json",
+      check: "String",
+      nullable: false
+    }
+  },
+  
+  members: {
+    __configPromise: null,
+    __libraryConfigs: null,
+    
+    /**
+     * Loads the configuration data
+     */
+    async load() {
+      let compileJsonPath = path.join(this.getRootDir(), this.getConfigFilename());
```

Comment:
same here: we have a lot of code duplication. See https://github.com/qooxdoo/qooxdoo-compiler/blob/master/source/class/qx/tool/config/Abstract.js

Reply:
The only duplication is that the default implementation loads a file via a call to a utility function - `qx.tool.config.Abstract` has a *lot* more code in it which does not apply here